internal/install/recipes/local_recipe_fetcher.go

```diff
@@ -0,0 +1,112 @@
+package recipes
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type LocalRecipeFetcher struct {
+	Path string
+}
+
+func (f *LocalRecipeFetcher) FetchRecipe(ctx context.Context, manifest *types.DiscoveryManifest, friendlyName string) (*types.Recipe, error) {
+	recipes, err := f.FetchRecommendations(ctx, manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, recipe := range recipes {
+		if recipe.Name == friendlyName {
+			return &recipe, nil
+		}
+	}
+
+	return nil, fmt.Errorf("%s: %w", friendlyName, ErrRecipeNotFound)
+}
+
+func (f *LocalRecipeFetcher) FetchRecommendations(ctx context.Context, manifest *types.DiscoveryManifest) ([]types.Recipe, error) {
+	recipes, err := f.FetchRecipes(ctx, manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	return manifest.ConstrainRecipes(recipes), nil
+}
+
+func (f *LocalRecipeFetcher) FetchRecipes(ctx context.Context, manifest *types.DiscoveryManifest) ([]types.Recipe, error) {
+	var recipes []types.Recipe
+	var err error
+
+	if f.Path == "" {
+		return nil, fmt.Errorf("unable to load recipes from empty path spec")
+	}
+
+	recipes, err = loadRecipesFromDir(ctx, f.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	return recipes, nil
+}
+
+func loadRecipesFromDir(ctx context.Context, path string) ([]types.Recipe, error) {
+	recipePaths := []string{}
+
+	log.WithFields(log.Fields{
+		"path": path,
+	}).Debug("loading recipes")
+
+	re := regexp.MustCompile(`\.ya?ml`)
+
+	err := filepath.Walk(
+		path,
+		func(path string, info os.FileInfo, err error) error {
+			if re.MatchString(path) {
```

Comment:
Not that it's a big deal, but would checking the file's extension directly with Go using `filepath.Ext(path)` make things "safer"? (i.e. "safer" for devs and inline readability and not relying on the wonderful mysterious world of regex 🦙 😉 ).